test/Authorization.t.sol

```diff
@@ -0,0 +1,305 @@
+        vault.setSsrBounds(1e27 - 1, FOUR_PCT_SSR);
+
+        vault.setSsrBounds(1e27, FOUR_PCT_SSR);
+    }
+
+    function test_setSsrBounds_aboveMaxSsrBoundary() public {
+        vm.startPrank(admin);
+        vm.expectRevert("Vault/ssr-too-high");
+        vault.setSsrBounds(1e27, MAX_SSR + 1);
+
+        vault.setSsrBounds(1e27, MAX_SSR);
+    }
+
+}
+
+contract SparkVaultSetSsrBoundsSuccessTests is SparkVaultTestBase {
+
+    event SsrBoundsSet(uint256 oldMinSsr, uint256 oldMaxSsr, uint256 newMinSsr, uint256 newMaxSsr);
+
+    function test_setSsrBounds() public {
+        assertEq(vault.minSsr(), 1e27);
+        assertEq(vault.maxSsr(), 1e27);
+
+        vm.startPrank(admin);
+        vm.expectEmit(address(vault));
+        emit SsrBoundsSet(1e27, 1e27, ONE_PCT_SSR, FOUR_PCT_SSR);
+        vault.setSsrBounds(ONE_PCT_SSR, FOUR_PCT_SSR);
+
+        assertEq(vault.minSsr(), ONE_PCT_SSR);
+        assertEq(vault.maxSsr(), FOUR_PCT_SSR);
+    }
+
+}
+
+contract SparkVaultRoleManagementFailureTests is SparkVaultTestBase {
+
+    function test_grantRole_notAdmin() public {
+        // > Check for DEFAULT_ADMIN_ROLE, SETTER_ROLE, TAKER_ROLE
+        bytes32[] memory roles = new bytes32[](3);
+        roles[0] = DEFAULT_ADMIN_ROLE;
+        roles[1] = SETTER_ROLE;
+        roles[2] = TAKER_ROLE;
+
+        for (uint256 i = 0; i < roles.length; i++) {
+            bytes32 role = roles[i];
+            vm.expectRevert(abi.encodeWithSignature(
+                "AccessControlUnauthorizedAccount(address,bytes32)",
+                address(this),
+                DEFAULT_ADMIN_ROLE
+            ));
+            // >> Action
+            vault.grantRole(role, address(0x1234));
+        }
+    }
+
+    function test_revokeRole_notAdmin() public {
+        // > Check for DEFAULT_ADMIN_ROLE, SETTER_ROLE, TAKER_ROLE
+        bytes32[] memory roles = new bytes32[](3);
+        roles[0] = DEFAULT_ADMIN_ROLE;
+        roles[1] = SETTER_ROLE;
+        roles[2] = TAKER_ROLE;
+
+        for (uint256 i = 0; i < roles.length; i++) {
+            bytes32 role = roles[i];
+            vm.expectRevert(abi.encodeWithSignature(
+                "AccessControlUnauthorizedAccount(address,bytes32)",
+                address(this),
+                DEFAULT_ADMIN_ROLE
+            ));
+            // >> Action
+            vault.revokeRole(role, address(0x1234));
+        }
+    }
+
+}
+
+contract SparkVaultRoleManagementSuccessTests is SparkVaultTestBase {
+
+    event RoleGranted(bytes32 indexed role, address indexed account, address indexed sender);
+    event RoleRevoked(bytes32 indexed role, address indexed account, address indexed sender);
+
+    function test_grantRole() public {
+        bytes32[] memory roles = new bytes32[](3);
+        roles[0] = DEFAULT_ADMIN_ROLE;
+        roles[1] = SETTER_ROLE;
+        roles[2] = TAKER_ROLE;
+
+        // > admin (DEFAULT_ADMIN_ROLE) should be allowed to grant DEFAULT_ADMIN_ROLE, SETTER_ROLE,
```

Comment:
Remove all `>`, this can be `// DEFAULT_ADMIN_ROLE should be able to grant DEFAULT_ADMIN_ROLE, SETTER_ROLE, TAKER_ROLE.`